XamarinCommunityToolkit/Behaviors/MatchTypes/HashtagMatchType.shared.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using Xamarin.Forms;
+namespace Microsoft.Toolkit.Xamarin.Forms.Behaviors
+{
+    public class HashtagMatchType : MatchType
+    {
+        public override Lazy<Regex> Regex { get; } = new Lazy<Regex>(() => new Regex(@"#\w+", RegexOptions.Singleline));
```

Comment:
I think you missed the `Compiled` flag, for regex

```suggestion
        public override Lazy<Regex> Regex { get; } = new Lazy<Regex>(() => new Regex(@"#\w+", RegexOptions.Singleline | RegexOptions.Compiled));
```